### DISCUSSION TITLE
What does parser type mean?

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#7743] [Q&A] [answered]
```
exports.types = {block:true, pragma:true};
```

I can't find it in dev doc

**Comment by linonetwo** [comment]
Sorry, it is under https://tiddlywiki.com/dev/#WikiRuleModules

It is under "module type" topic, not "parser" topic.

But I'm still not sure, is `block` mean only work for multi-line text, and `inline` mean only work for single line text? 

https://github.com/Jermolene/TiddlyWiki5/blob/49c96901f355de025a03a9ede3b4974fc8d5fc9e/core/modules/parsers/wikiparser/rules/transcludeblock.js#L6-L11

https://github.com/Jermolene/TiddlyWiki5/blob/49c96901f355de025a03a9ede3b4974fc8d5fc9e/core/modules/parsers/wikiparser/rules/transcludeinline.js#L6-L11

I don't see any difference.

**Comment by saqimtiaz** [comment, marked as answer]
@linonetwo this might be helpful: https://tiddlywiki.com/#WikiText%20Parser%20Modes

**Reply by linonetwo** [reply]
Thanks, so the criteria is end with something.
